lib/consul/async/consul_template_engine.rb

```diff
@@ -61,15 +61,18 @@ def do_run(template_manager, template_renders)
         STDERR.puts "[FATAL]#{e}"
         template_manager.terminate
         EventMachine.stop
+        return 1
       rescue StandardError => e
         STDERR.puts "[FATAL] Error occured: #{e.inspect} - #{e.backtrace.join("\n\t")}"
         template_manager.terminate
         EventMachine.stop
+        return 2
       end
 
       # Run template engine as fast as possible until first rendering occurs
       def do_run_fast(template_manager, template_renders)
-        do_run(template_manager, template_renders)
+        s = do_run(template_manager, template_renders)
+        return s unless s.zero?
 
         return if @all_templates_rendered || @periodic_started
```

Comment:
it does not fail as it should it failing in 
```
           EventMachine.next_tick do
             do_run_fast(template_manager, template_renders)
           end
```

Reply:
ok, will use a shared object in the engine Class
I don't specially like this, but I don't see how we can make EventMachine inside the run_loop

The only think we probably don't really mind right now, is that it may result in a race condition
I don't know EM enough to know if it will happens, but I think we're safe